src/group_theory/order_of_element.lean

```diff
@@ -235,15 +364,26 @@ begin
   { apply order_of_dvd_of_pow_eq_one,
     rw [← pow_mul, ← nat.mul_div_assoc _ (gcd_dvd_left _ _), mul_comm,
         nat.mul_div_assoc _ (gcd_dvd_right _ _), pow_mul, pow_order_of_eq_one, one_pow] },
-  { have gcd_pos : 0 < gcd (order_of a) n, exact gcd_pos_of_pos_right _ (nat.pos_of_ne_zero h),
+  { have gcd_pos : 0 < gcd (order_of a) n := gcd_pos_of_pos_right _ (nat.pos_of_ne_zero h),
     apply coprime.dvd_of_dvd_mul_right (coprime_div_gcd_div_gcd gcd_pos),
     apply dvd_of_mul_dvd_mul_right gcd_pos,
     rw [nat.div_mul_cancel (gcd_dvd_left _ _), mul_assoc, nat.div_mul_cancel (gcd_dvd_right _ _),
-    mul_comm],
+        mul_comm],
     apply order_of_dvd_of_pow_eq_one,
     rw [pow_mul, pow_order_of_eq_one] }
 end
 
+variables (x)
+
+lemma add_order_of_nsmul' (h : n ≠ 0) :
+  add_order_of (n •ℕ x) = add_order_of x / gcd (add_order_of x) n :=
+begin
+  repeat {rw ← order_of_of_add_eq_add_order_of},
+  rwa [of_add_nsmul, order_of_pow'],
+end
```

Comment:
```suggestion
lemma add_order_of_nsmul' (h : n ≠ 0) :
  add_order_of (n •ℕ x) = add_order_of x / gcd (add_order_of x) n :=
by simpa [← order_of_of_add_eq_add_order_of, of_add_nsmul] using order_of_pow' _ h
```

and similarly in other places where you have `repeat {rw ←}`, if possible. There might be some helper simp lemmas for those other cases.